### PR TITLE
issue #439 - added a script for detecting the country of the phone number 

### PR DESCRIPTION
**Pull request by Manav958**
I have added the feature to detect the country from the given phone numbers
using pycountry and phonenumbers i have tested it with some numbers from various countries it works good.
Please accept the PR or suggest the required changes.
Thankyou

To Run it :-
configure the GUI part and run the python script named number_mask.py

** I had to upgrade the version of networkx from exactly 2.6 to >= 2.6

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ gitignore-parser = "==0.1.1"
 kthread = "==0.2.3"
 markdown = "==3.4.4"
 marshmallow = "==3.20.1"
-networkx = "==2.6"
+networkx = ">=2.6"
```
Comment (FredLL-Avaiga) [inline Pipfile:21]: this should be discussed by the team

File: frontend/taipy-gui/package.json
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/package.json:1]: You  should have to modify this file

File: frontend/taipy/package.json
Comment (FredLL-Avaiga) [inline frontend/taipy/package.json:1]: you shouldn't commit this file

File: taipy/_entrypoint.py
Comment (FredLL-Avaiga) [inline taipy/_entrypoint.py:1]: no need to commit this file

File: Pipfile (hunk 1 above)
Comment (Manav958) [reply Pipfile:21]: Ok
I did it because it was producing error while installing packages.

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
You are not proposing any change to the code base.
this issue should have an impact on Input.tsx

File: Pipfile (hunk 1 above)
Comment (FredLL-Avaiga) [reply Pipfile:21]: then raise an issue

File: frontend/taipy-gui/package.json
Comment (Manav958) [reply frontend/taipy-gui/package.json:1]: Alright , I am doing the necessary changes.

File: Pipfile (hunk 1 above)
Comment (Manav958) [reply Pipfile:21]: Hi a Quick question
should i modify Input.tsx to have a field for phone number and then trigger the py script for validation
cause i currently dont see that

please guide me through this . 
if i am wrong please tell exactly what is required.
Thankyou

** pardon me for asking a lot of questions as this is my first time contributing to a big organization.

Comment (FredLL-Avaiga) [reply Pipfile:21]: the aim of the game is indeed to modify Input.tsx to support masks (phone number but maybe something else ...).
Once your modification is done and you've wrote frontend tests in Input.spec.tsx, you build the bundle with `npm run build:dev`
then you create a script that will allow you to test the new features you've added
then you create a few examples in doc/gui/examples/

Comment (FredLL-Avaiga) [reply Pipfile:21]: PS: And you should update your branch

Comment (Manav958) [reply Pipfile:21]: Hi ,
I had to create a backend URL in order to expose the function of phone number masking
where should i keep that file current its in the root.

also i have updated the Input.tsx file's props for calling the backend

Comment (FredLL-Avaiga) [reply Pipfile:21]: I would really prefer a frontend only solution

Comment (Manav958) [reply Pipfile:21]: Oh Okay 
I will do that

Comment (Manav958) [reply Pipfile:21]: Hi sir
Sorry for delayed reply i fell sick

i have a few questions
I implemented the logic in front but not sure how to test that
The logic takes the phone number and slice the last four digits and mask the remaining ones as "*" 

Could help me? Understanding what steps must be taken

Comment (Manav958) [reply Pipfile:21]: Also while running npm run build:dev it throws error

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory

Comment (FredLL-Avaiga) [reply Pipfile:21]: we use `NODE_OPTIONS: --max-old-space-size=4096`

**Comment by FredLL-Avaiga** [comment]
Are you still working on this PR @Manav958  ?
If you are, please resolve the conflicts, update your branch and answer the questions
If you are not, please close the PR

**Pull request closed without merging**